src/System.Text.Json/tests/JsonArrayTests.cs

```diff
@@ -0,0 +1,344 @@
+            Assert.IsType<JsonArray>(innerJsonArray[0]);
+        }
+
+        [Fact]
+        public static void TestCreatingJsonArrayFromCollection()
+        {
+            var employeesIds = new JsonArray(EmployeesDatabase.GetTenBestEmployees().Select(employee => new JsonString(employee.Key)));
+
+            JsonString prevId = new JsonString();
+            foreach (JsonNode employeeId in employeesIds)
+            {
+                Assert.IsType<JsonString>(employeeId);
+                var employeeIdString = employeeId as JsonString;
+                Assert.NotEqual(prevId, employeeIdString);
+                prevId = employeeIdString;
+            }
+        }
+
+        [Fact]
+        public static void TestCreatingJsonArrayFromCollectionOfString()
+        {
+            var employeesIds = new JsonArray(EmployeesDatabase.GetTenBestEmployees().Select(employee => employee.Key));
+
+            JsonString prevId = new JsonString();
+            foreach (JsonNode employeeId in employeesIds)
+            {
+                Assert.IsType<JsonString>(employeeId);
+                var employeeIdString = employeeId as JsonString;
+                Assert.NotEqual(prevId, employeeIdString);
+                prevId = employeeIdString;
+            }
+        }
+
+        [Fact]
+        public static void TestAddingToJsonArray()
+        {
+            var employeesIds = new JsonArray();
+
+            foreach (KeyValuePair<string, JsonNode> employee in EmployeesDatabase.GetTenBestEmployees())
+            {
+                employeesIds.Add(employee.Key);
+            }
+
+            JsonString prevId = new JsonString();
+            foreach (JsonNode employeeId in employeesIds)
+            {
+                Assert.IsType<JsonString>(employeeId);
+                var employeeIdString = employeeId as JsonString;
+                Assert.NotEqual(prevId, employeeIdString);
+                prevId = employeeIdString;
+            }
+        }
+
+        [Fact]
+        public static void TestAccesingNestedJsonArrayGetPropertyMethod()
+        {
+            var issues = new JsonObject()
+            {
+                { "features", new JsonString [] { "new functionality 1", "new functionality 2" } },
+                { "bugs", new JsonString [] { "bug 123", "bug 4566", "bug 821" } },
+                { "tests", new JsonString [] { "code coverage" } },
+            };
+
+            issues.GetJsonArrayProperty("bugs").Add("bug 12356");
+            ((JsonString)issues.GetJsonArrayProperty("features")[0]).Value = "feature 1569";
+            ((JsonString)issues.GetJsonArrayProperty("features")[1]).Value = "feature 56134";
+
+            Assert.True(((JsonArray)issues["bugs"]).Contains("bug 12356"));
+            Assert.Equal("feature 1569", (JsonString)((JsonArray)issues["features"])[0]);
+            Assert.Equal("feature 56134", (JsonString)((JsonArray)issues["features"])[1]);
+        }
+    }
```

Comment:
* There are a lot of ctors, each one needs to be tested.
* All of the tests are homogeneous (always numbers or always strings), some heterogeneous tests should exist.
* Combine using the enumerable ctors and the add methods instead of always doing one or the other.